### PR TITLE
Fix for ScriptEngines not being released upon AC shutdown

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
https://github.com/highfidelity/hifi/pull/13536

This earlier PR fixed a Windows crash when the Agent AC was restarted by delaying the destruction of the ScriptEngines until the AC ran as an Agent again. The current PR destroys the ScriptEngine immediately upon its takedown.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11853/HighFidelity-Beta-PR13755-5927188.apk

**Comment by SimonWalton-HiFi** [comment]
## Test Plan

Run the domain server, ACs & interface locally from the PR. 

- Go to the DS Nodes listing: http://localhost:40100 -> Nodes. 
- Restart the Agent AC by clicking on the Kill? X or clicking on Kill all Nodes. (NB: killing individual Nodes is currently nonoperational - https://highfidelity.manuscript.com/f/cases/17264/ .)

 To test operation of the Agent AC:

- Run the record script on Interface: Edit -> Running Scripts -> From Disk -> record.js.
- Run the recording-playback script on the Agent: http://localhost:40100 -> Content -> Scripts. Add a script and enter `https://content.highfidelity.com/Scripts/playRecordingAC.js` as the script URL. Set instances to 1.
- Click on Save and Restart.
- Record your avatar via Record (on menu bar) -> Record.
- Playback the avatar recording (should start automatically after recording is halted).

With this PR the Agent should restart without crashing and playback should work correctly.

See also Test Plan from previous PR: https://github.com/highfidelity/hifi/pull/13536#issuecomment-402264306

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11853/HighFidelity-Beta-PR13755-5927188.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11853/HighFidelity-Beta-Interface-PR13755-5927188.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11853/HighFidelity-Beta-PR13755-5927188.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11853/HighFidelity-Beta-Interface-PR13755-5927188.dmg

File: assignment-client/src/Agent.cpp [hunk 1]
```diff
@@ -497,7 +497,6 @@ void Agent::executeScript() {
     Frame::clearFrameHandler(AUDIO_FRAME_TYPE);
     Frame::clearFrameHandler(AVATAR_FRAME_TYPE);
 
-    DependencyManager::destroy<RecordingScriptingInterface>();
```
Comment (birarda) [inline assignment-client/src/Agent.cpp]: Hoping I didn't specifically put this here for a reason

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11919/HighFidelity-Beta-PR13755-df9c65e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11919/HighFidelity-Beta-PR13755-df9c65e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11919/HighFidelity-Beta-Interface-PR13755-df9c65e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11919/HighFidelity-Beta-PR13755-df9c65e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11919/HighFidelity-Beta-Interface-PR13755-df9c65e.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Agent responds well.

**Comment by conklin94122** [comment]
build this please

**Comment by conklin94122** [comment]
QA Approval is stale, retesting.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12141/HighFidelity-Beta-PR13755-df9c65e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12141/HighFidelity-Beta-PR13755-df9c65e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12141/HighFidelity-Beta-Interface-PR13755-df9c65e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12141/HighFidelity-Beta-PR13755-df9c65e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12141/HighFidelity-Beta-Interface-PR13755-df9c65e.exe

**Comment by qlipdahlin** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
@conklin94122 PASS

Using the PR13536 Test Plan Supplement, I downloaded and adjust the .bat file to work with PR13755 and tested the Agent nodes with the build. Due to the difficulties with killing individual nodes, kill all was the best alternative. Agent did not crash.

Ran the record.js and ran the playRecordingAC.js on the server and the application and recording worked properly.

**Pull request merged**